feat: add C implementation for `math/base/special/gamma-lanczos-sum-expg-scaled`

gunjjoshi opened the pull request:
Resolves #1835 

## Description

> What is the purpose of this pull request?

This pull request:

-   adds C implementation for [`math/base/special/gamma-lanczos-sum-expg-scaled`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/gamma-lanczos-sum-expg-scaled).
## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves a part of #649 
-   fixes #1835 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   ☑️ Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

gunjjoshi committed:
feat: add C implementation for math/base/special/gamma-lanczos-sum-expg-scaled

gunjjoshi committed:
Update README.md

Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>

gunjjoshi committed:
Update benchmark.native.js

Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>

gunjjoshi committed:
Update main.c

Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>

gunjjoshi committed:
Update test.native.js

Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/gamma-lanczos-sum-expg-scaled/src/main.c:
```diff
@@ -0,0 +1,74 @@
+	} else {
+		ax = x;
+	}
+	if ( ax <= 1.0 ) {
```
Same comment as in https://github.com/stdlib-js/stdlib/pull/1834#discussion_r1520513679

kgryte replied:
If the issue is L58, we should update `evalrational-compile-c` to handle infinities.

gunjjoshi replied:
Thanks for these suggestions. Once we are done with the PR for [`gamma-lanczos-sum`](https://github.com/stdlib-js/stdlib/pull/1834), then I will make changes according to that here, so that things aren't repetitive.

Snehil-Shah committed:
feat: add C implementation for `math/base/special/acsch`

Closes: https://github.com/stdlib-js/stdlib/issues/860
PR-URL: 	https://github.com/stdlib-js/stdlib/pull/1833
Co-authored-by: Athan Reines <kgryte@gmail.com>
Reviewed-by: Athan Reines <kgryte@gmail.com> 
Signed-off-by: Snehil Shah <130062020+Snehil-Shah@users.noreply.github.com>
Signed-off-by: Athan Reines <kgryte@gmail.com>

Planeshifter committed:
build: run C benchmarks in CI workflow

Userf1995 committed:
feat: add `math/base/assert/is-nonnegative-finite`

PR-URL: #1829
Closes: #1828

saisrikardumpeti committed:
feat: add `math/base/special/asecd`

PR-URL: #1788
Closes: #41

---------

Signed-off-by: Sai Srikar Dumpeti <80447788+the-r3aper7@users.noreply.github.com>
Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

Jai0401 committed:
docs: improve README examples for `stats/base/dists/rayleigh` namespace

PR-URL: #1754
Closes: #1642

---------

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

aman-095 committed:
feat: add C implementation for `math/base/special/log2`

PR-URL: #1717

---------

Signed-off-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Signed-off-by: Athan Reines <kgryte@gmail.com>
Co-authored-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Co-authored-by: Athan Reines <kgryte@gmail.com>
Reviewed-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Reviewed-by: Athan Reines <kgryte@gmail.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

Userf1995 committed:
docs: improve README examples of `stats/base/dists/negative-binomial`

PR-URL: #1774
Closes: #1637

---------

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

stdlib-bot committed:
docs: update list of contributors

PR-URL: #1837

Signed-off-by: stdlib-bot <82920195+stdlib-bot@users.noreply.github.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

stdlib-bot committed:
docs: update related packages sections 

PR-URL: #1838
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

gunjjoshi committed:
feat: add C implementation for `math/base/special/gammasgn`

PR-URL: #1841
Closes: #649

---------

Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>
Signed-off-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Co-authored-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Reviewed-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>

Userf1995 committed:
feat: add C implementation for `math/base/special/ahaversin`

PR-URL: #1840
Closes: #1420

---------

Signed-off-by: Rejoan Sardar <119718513+Rejoan-Sardar@users.noreply.github.com>
Signed-off-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Co-authored-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Reviewed-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

performant23 committed:
feat: add C implementation for `math/base/special/xlogy`

PR-URL: #1827
Closes: #1815

---------

Signed-off-by: Rutam <138517416+performant23@users.noreply.github.com>
Signed-off-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Co-authored-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Reviewed-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

saisrikardumpeti committed:
feat: add `math/base/special/cscd`

PR-URL: #1786
Closes: #39

---------

Signed-off-by: Sai Srikar Dumpeti <80447788+the-r3aper7@users.noreply.github.com>
Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
Co-authored-by: stdlib-bot <82920195+stdlib-bot@users.noreply.github.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

saisrikardumpeti committed:
feat: add `math/base/special/acotd`

PR-URL: #1785
Closes: #43

---------

Signed-off-by: Sai Srikar Dumpeti <80447788+the-r3aper7@users.noreply.github.com>
Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

saisrikardumpeti committed:
feat: add `math/base/special/acscd`

PR-URL: #1789
Closes: #42

---------

Signed-off-by: Sai Srikar Dumpeti <80447788+the-r3aper7@users.noreply.github.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

saisrikardumpeti committed:
feat: add `math/base/special/secd`

PR-URL: #1787
Closes: #38

---------

Signed-off-by: Sai Srikar Dumpeti <80447788+the-r3aper7@users.noreply.github.com>
Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Co-authored-by: stdlib-bot <82920195+stdlib-bot@users.noreply.github.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

stdlib-bot committed:
docs: update REPL namespace documentation

PR-URL: 	https://github.com/stdlib-js/stdlib/pull/1843
Reviewed-by: Athan Reines <kgryte@gmail.com>
Signed-off-by: stdlib-bot <82920195+stdlib-bot@users.noreply.github.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>

saisrikardumpeti committed:
feat: add `math/base/special/acosd`

PR-URL: #1782 
Closes: #36

---------

Signed-off-by: Sai Srikar Dumpeti <80447788+the-r3aper7@users.noreply.github.com>
Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

saisrikardumpeti committed:
feat: add `math/base/special/atand`

PR-URL: #1784
Closes: #37

---------

Signed-off-by: Sai Srikar Dumpeti <80447788+the-r3aper7@users.noreply.github.com>
Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter committed:
build: delete gypfile field when files are not present

saisrikardumpeti committed:
feat: add `math/base/special/asind`

PR-URL: #1781
Closes: #35

---------

Signed-off-by: Sai Srikar Dumpeti <80447788+the-r3aper7@users.noreply.github.com>
Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

stdlib-bot committed:
chore: update package meta data

PR-URL: #1856

Signed-off-by: stdlib-bot <82920195+stdlib-bot@users.noreply.github.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

Userf1995 committed:
docs: fix JSDoc description in `math/base/special/acoversin`

PR-URL: #1853

Signed-off-by: Rejoan Sardar <119718513+Rejoan-Sardar@users.noreply.github.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

adityacodes30 committed:
feat: add `string/base/last-code-point`

PR-URL: #1755
Closes: #1702

---------

Signed-off-by: Aditya Sapra <110766802+adityacodes30@users.noreply.github.com>
Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
Co-authored-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

stdlib-bot committed:
docs: update related packages sections

PR-URL: #1857

Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

kgryte committed:
docs: update contributors

kgryte committed:
fix: update dependencies

kgryte committed:
fix: use correct package names

kgryte committed:
fix: update include path

kgryte committed:
fix: update build configurations

performant23 committed:
refactor: update `blas/ext/base/drev` to follow current project conventions

Closes: https://github.com/stdlib-js/stdlib/issues/1488
PR-URL: 	https://github.com/stdlib-js/stdlib/pull/1839
Ref: https://github.com/stdlib-js/stdlib/issues/1152
Co-authored-by: Athan Reines <kgryte@gmail.com>
Reviewed-by: Athan Reines <kgryte@gmail.com> 
Signed-off-by: Pranav Goswami <goswami.4@iitj.ac.in>
Signed-off-by: Athan Reines <kgryte@gmail.com>
Co-authored-by: Pranav Goswami <goswami.4@iitj.ac.in>

kgryte committed:
fix: update build configurations

kgryte committed:
fix: update build configurations

kgryte committed:
fix: update build configurations

kgryte committed:
fix: update build configurations and fix argument extraction

kgryte committed:
fix: update build configurations

stdlib-bot committed:
docs: update REPL namespace documentation

PR-URL: 	https://github.com/stdlib-js/stdlib/pull/1858
Co-authored-by: Athan Reines <kgryte@gmail.com>
Reviewed-by: Athan Reines <kgryte@gmail.com> 
Signed-off-by: stdlib-bot <82920195+stdlib-bot@users.noreply.github.com>

gunjjoshi committed:
feat: add C implementation for `math/base/special/bernoulli`

PR-URL: #1860 
Closes: #1664

---------

Signed-off-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>
Co-authored-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Reviewed-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>

AuenKr committed:
refactor: update `blas/ext/base/dssum` to follow current projects conventions

PR-URL: #1861
Closes: #1501

---------

Signed-off-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Co-authored-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Reviewed-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>

Userf1995 committed:
feat: add C implementation for `math/base/assert/int32-is-odd`

PR-URL: #1865
Closes: #1806 

---------

Signed-off-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
Co-authored-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

aman-095 committed:
feat: add C implementation for `math/base/assert/is-safe-integer`

PR-URL: #1862

---------

Signed-off-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Co-authored-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Reviewed-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

performant23 committed:
feat: add C implementation for `math/base/assert/is-nonpositive-finite`

PR-URL: #1859 
Closes: #1831

---------

Signed-off-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Co-authored-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Reviewed-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

stdlib-bot committed:
docs: update REPL namespace documentation

PR-URL: #1877

Signed-off-by: stdlib-bot <82920195+stdlib-bot@users.noreply.github.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

Userf1995 committed:
feat: add C implementation for `math/base/special/xlog1py`

PR-URL: #1847 
Closes: #1842

---------

Signed-off-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
Co-authored-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Pranav <85227306+Pranavchiku@users.noreply.github.com> 
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

gunjjoshi committed:
feat: add C implementation for `math/base/special/boxcox`

PR-URL: 	https://github.com/stdlib-js/stdlib/pull/1870
Ref: https://github.com/stdlib-js/stdlib/issues/649
Co-authored-by: Athan Reines <kgryte@gmail.com>
Reviewed-by: Athan Reines <kgryte@gmail.com> 
Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>
Signed-off-by: Athan Reines <kgryte@gmail.com>

Userf1995 committed:
feat: add C implementation for `math/base/special/boxcoxinv`

PR-URL: 	https://github.com/stdlib-js/stdlib/pull/1878
Ref: https://github.com/stdlib-js/stdlib/issues/649
Co-authored-by: Athan Reines <kgryte@gmail.com>
Reviewed-by: Athan Reines <kgryte@gmail.com> 
Signed-off-by: Rejoan Sardar <119718513+Rejoan-Sardar@users.noreply.github.com>
Signed-off-by: Athan Reines <kgryte@gmail.com>

gunjjoshi committed:
feat: add C implementation for `math/base/special/boxcox1p`

PR-URL: #1883

---------

Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>
Signed-off-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Co-authored-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Reviewed-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>

Userf1995 committed:
feat: add C implementation for `math/base/special/boxcox1pinv`

PR-URL: #1885

---------

Signed-off-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Co-authored-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Reviewed-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>

performant23 committed:
feat: add C implementation for `math/base/assert/uint32-is-pow2`

PR-URL: #1886
Closes: #1868

---------

Signed-off-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
Co-authored-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter committed:
chore: remove indentation

Planeshifter committed:
build: add workflow to auto-assign labels to PRs

Planeshifter committed:
build: rename labeler config file

Userf1995 committed:
docs: improve README examples of `stats/base/dists/truncated-normal`

PR-URL: #1872
Cloes: #1647

---------

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter committed:
build: explicitly set token used in action

saisrikardumpeti committed:
feat: add `math/base/special/tand`

PR-URL: #1771
Closes: #34

---------

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

saisrikardumpeti committed:
feat: add `math/base/special/cotd`

PR-URL: #1772
Closes: #40

---------

Signed-off-by: Sai Srikar Dumpeti <80447788+the-r3aper7@users.noreply.github.com>
Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

kgryte committed:
fix: remove labels

The previous configuration was generating false positives.

aman-095 committed:
feat: add C implementation for `math/base/assert/is-prime`

PR-URL: 	https://github.com/stdlib-js/stdlib/pull/1866
Co-authored-by: Athan Reines <kgryte@gmail.com>
Reviewed-by: Athan Reines <kgryte@gmail.com> 
Signed-off-by: Athan Reines <kgryte@gmail.com>

gunjjoshi committed:
feat: add C implementation for `math/base/special/acotd`

PR-URL: #1879 
Closes: #649

---------

Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>
Signed-off-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Co-authored-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Reviewed-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>

gunjjoshi committed:
feat: add C implementation for `math/base/special/asecd`

PR-URL: #1880

---------

Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>
Signed-off-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Co-authored-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Reviewed-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>

Userf1995 committed:
feat: add C implementation for `math/base/special/acsc`

PR-URL: #1887
Closes: #859

---------

Signed-off-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Co-authored-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Reviewed-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>

Userf1995 committed:
feat: add C implementation for `math/base/special/atand`

PR-URL: #1881

---------

Signed-off-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Co-authored-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Reviewed-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>

gunjjoshi committed:
feat: add C implementation for `math/base/special/acosd`

PR-URL: #1871

---------

Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>
Signed-off-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Co-authored-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Reviewed-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

performant23 committed:
feat: add C implementation for `math/base/special/factorial2`

PR-URL: #1898 
Closes: #1892

---------

Signed-off-by: Rutam <138517416+performant23@users.noreply.github.com>
Signed-off-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
Co-authored-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

kgryte committed:
docs: clean-up examples and descriptions

Ref: #1339

kgryte committed:
feat: add `countSameValue` to namespace

kgryte committed:
test: add test case to ensure full test coverage

Ref: #1339
Ref: https://github.com/stdlib-js/stdlib/commit/634044b1b369b08b2748c3f6f457c2b1259ccfa3

kgryte committed:
docs: add notes concerning algorithm

Ref: #1339
Ref: https://github.com/stdlib-js/stdlib/commit/634044b1b369b08b2748c3f6f457c2b1259ccfa3

kgryte committed:
docs: add same value example

Ref: #1339
Ref: https://github.com/stdlib-js/stdlib/commit/634044b1b369b08b2748c3f6f457c2b1259ccfa3

kgryte committed:
docs: clean-up documentation and examples and rename folder

Ref: #1384
Ref: https://github.com/stdlib-js/stdlib/commit/1ed81b78fd6469d374c7b033f258e709934539b2

kgryte committed:
feat: add `countSameValueZero` to namespace

kgryte committed:
docs: fix copy

stdlib-bot committed:
docs: update related packages sections 

PR-URL: #1907

Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

kgryte committed:
docs: update test messages, comments, and benchmarks

Ref: #1354
Ref: https://github.com/stdlib-js/stdlib/commit/609682c8ef64ecf0fb6a4d7094e226a99a985ab5

kgryte committed:
feat: add `isNonNegativeFinite` to namespace

kgryte committed:
feat: add `isNonNegativeFinite` to namespace

gunjjoshi committed:
feat: add C implementation for `math/base/special/fast/hypot` 

PR-URL: #1910

---------

Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>
Reviewed-by: Pranav Goswami <goswami.4@iitj.ac.in>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

gunjjoshi committed:
feat: add C implementation for `math/base/special/fast/acosh`

PR-URL: #1897

---------

Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>
Co-authored-by:  Pranav Goswami <goswami.4@iitj.ac.in>
Reviewed-by: Pranav Goswami <goswami.4@iitj.ac.in>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

gunjjoshi committed:
feat: add C implementation for `math/base/special/fast/abs`

PR-URL: #1894

---------

Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>
Co-authored-by: Athan Reines <kgryte@gmail.com>
Reviewed-by: Pranav Goswami <goswami.4@iitj.ac.in>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

stdlib-bot committed:
docs: update namespace table of contents

PR-URL: #1920

Signed-off-by: stdlib-bot <82920195+stdlib-bot@users.noreply.github.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

Userf1995 committed:
feat: add C implementation for `math/base/special/asind`

PR-URL: #1876

---------

Signed-off-by: Rejoan Sardar <119718513+Rejoan-Sardar@users.noreply.github.com>
Reviewed-by: Pranav Goswami <goswami.4@iitj.ac.in>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

stdlib-bot committed:
docs: update namespace table of contents

PR-URL: #1921

Signed-off-by: stdlib-bot <82920195+stdlib-bot@users.noreply.github.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com> 
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

Userf1995 committed:
feat: add C implementation for `math/base/special/fast/asinh`

PR-URL: #1908

---------

Signed-off-by: Rejoan Sardar <119718513+Rejoan-Sardar@users.noreply.github.com>
Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter committed:
fix: move addition by one into the sqrt invocation

Planeshifter committed:
build: update check given leading stdlib prefix is already included

Planeshifter committed:
docs: clean-up C function parameter and return annotations

Planeshifter committed:
docs: fix function names in C example code

stdlib-bot committed:
chore: update package meta data

PR-URL: #1926

Signed-off-by: stdlib-bot <82920195+stdlib-bot@users.noreply.github.com>
Co-authored-by: Athan Reines <kgryte@gmail>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

AuenKr committed:
refactor: update `blas/ext/base/dnansum` to follow current project conventions

PR-URL: #1899
Closes: #1483

---------

Signed-off-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Co-authored-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Reviewed-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>

stdlib-bot committed:
docs: update REPL namespace documentation

PR-URL: #1928

Signed-off-by: stdlib-bot <82920195+stdlib-bot@users.noreply.github.com>
Co-authored-by: Planeshifter <1913638+Planeshifter@users.noreply.github.com>
Reviewed-by: Planeshifter <1913638+Planeshifter@users.noreply.github.com>

Userf1995 committed:
feat: add C implementation for `math/base/special/fast/max`

PR-URL: #1912

---------

Signed-off-by: Pranav Goswami <goswami.4@iitj.ac.in>
Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
Co-authored-by: Pranav Goswami <goswami.4@iitj.ac.in>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Pranav Goswami <goswami.4@iitj.ac.in>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

Userf1995 committed:
feat: add C implementation for `math/base/special/fast/min`

PR-URL: #1913
Closes: #1911

---------

Signed-off-by: Rejoan Sardar <119718513+Rejoan-Sardar@users.noreply.github.com>
Signed-off-by: Pranav Goswami <goswami.4@iitj.ac.in>
Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
Co-authored-by: Pranav Goswami <goswami.4@iitj.ac.in>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Pranav Goswami <goswami.4@iitj.ac.in>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

stdlib-bot committed:
docs: update related packages sections

PR-URL: #1930

---------

Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

performant23 committed:
fix: update variable type declarations in `math/base/assert/int32-is-odd`

PR-URL: #1935

Reviewed-by: Pranav Goswami <goswami.4@iitj.ac.in>

Rec0iL99 committed:
feat: add `utils/any-own-by`

PR-URL: #1851
Closes: #819

---------

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

stdlib-bot committed:
feat: update namespace TypeScript declarations

PR-URL: #1937

Signed-off-by: stdlib-bot <82920195+stdlib-bot@users.noreply.github.com>
Reviewed-by: Planeshifter <1913638+Planeshifter@users.noreply.github.com>

gunjjoshi committed:
feat: add C implementation for `math/base/special/modf`

PR-URL:  #1934
Closes: #1916

---------

Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>
Signed-off-by: Pranav Goswami <goswami.4@iitj.ac.in>
Co-authored-by: Pranav Goswami <goswami.4@iitj.ac.in>
Reviewed-by: Pranav Goswami <goswami.4@iitj.ac.in>

gunjjoshi committed:
feat: add C implementation for `math/base/special/fast/pow-int`

PR-URL: #1919
Closes: #1917

---------

Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>
Signed-off-by: Pranav Goswami <goswami.4@iitj.ac.in>
Co-authored-by: Pranav Goswami <goswami.4@iitj.ac.in>
Reviewed-by: Pranav Goswami <goswami.4@iitj.ac.in>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

saisrikardumpeti committed:
feat: add `math/base/special/cosd`

PR-URL: #1777
Closes: #33

---------

Signed-off-by: Sai Srikar Dumpeti <80447788+the-r3aper7@users.noreply.github.com>
Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

Userf1995 committed:
feat: add C implementation for `math/base/special/fast/atanh`

PR-URL: #1909

---------

Signed-off-by: Rejoan Sardar <119718513+Rejoan-Sardar@users.noreply.github.com>
Signed-off-by: Pranav Goswami <goswami.4@iitj.ac.in>
Co-authored-by:Pranav Goswami <goswami.4@iitj.ac.in>
Reviewed-by: Pranav Goswami <goswami.4@iitj.ac.in>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

performant23 committed:
feat: add C implementation for `math/base/assert/is-composite`

PR-URL: #1918
Closes: #1901

---------

Signed-off-by: Rutam <138517416+performant23@users.noreply.github.com>
Signed-off-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
Co-authored-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Pranav Goswami <goswami.4@iitj.ac.in>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter committed:
build: update installed R version

R linting seems to be broken on CI, see `lint_random_files.yml` workflow
run such as the following:

https://github.com/stdlib-js/stdlib/actions/runs/8319437589/job/22762740425#step:15:988

This PR updates the installed R version to see if that resolves the
issue as all dependencies are correctly installed from CRAN locally
when ussing that version.

Planeshifter committed:
build: skip labeler for auto PRs and only assign when all files match

Planeshifter committed:
build: update R lint configuration

stdlib-bot committed:
docs: update list of contributors

PR-URL: #1940

Signed-off-by: stdlib-bot <82920195+stdlib-bot@users.noreply.github.com>
Reviewed-by: Planeshifter <1913638+Planeshifter@users.noreply.github.com>

stdlib-bot committed:
docs: update related packages sections

PR-URL: #1941

Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

AuenKr committed:
refactor: update `blas/ext/base/dcusum` to follow current project conventions

PR-URL: 	https://github.com/stdlib-js/stdlib/pull/1867
Closes: https://github.com/stdlib-js/stdlib/issues/1471
Ref: https://github.com/stdlib-js/stdlib/issues/1152
Co-authored-by: Athan Reines <kgryte@gmail.com>
Reviewed-by: Athan Reines <kgryte@gmail.com> 
Signed-off-by: Athan Reines <kgryte@gmail.com>

stdlib-bot committed:
docs: update REPL namespace documentation

PR-URL: #1956

Signed-off-by: stdlib-bot <82920195+stdlib-bot@users.noreply.github.com>
Co-authored-by: kgryte <2643044+kgryte@users.noreply.github.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

performant23 committed:
fix: missing headers in C examples

PR-URL: 	https://github.com/stdlib-js/stdlib/pull/1958
Reviewed-by: Athan Reines <kgryte@gmail.com>
Reviewed-by: Pranav Goswami <goswami.4@iitj.ac.in>

gunjjoshi committed:
feat: add `constants/float32/half-pi`

PR-URL: 	https://github.com/stdlib-js/stdlib/pull/1953
Ref: https://github.com/stdlib-js/stdlib/issues/1944
Co-authored-by: Athan Reines <kgryte@gmail.com>
Reviewed-by: Athan Reines <kgryte@gmail.com> 
Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>
Signed-off-by: Athan Reines <kgryte@gmail.com>

kgryte committed:
fix: update documentation to reflect correct seed range

The seed for MT19937 must be a nonzero positive integer, as discussed
in the original paper describing the Mersenne Twister PRNG.

Closes: https://github.com/stdlib-js/stdlib/issues/1963

kgryte committed:
docs: update contributing guide to explicitly require running `make install`

Closes: https://github.com/stdlib-js/stdlib/issues/1678

kgryte committed:
docs: add note regarding spaces in directory path

kgryte committed:
docs: add note concerning native add-ons

kgryte committed:
docs: add link

gunjjoshi committed:
refactor: add support for ratios evaluating as infinity in `math/base/tools/evalrational-compile-c`

PR-URL: 	https://github.com/stdlib-js/stdlib/pull/1970
Ref: https://github.com/stdlib-js/stdlib/pull/1834#discussion_r1531496093
Co-authored-by: Athan Reines <kgryte@gmail.com>
Reviewed-by: Athan Reines <kgryte@gmail.com> 
Signed-off-by: Athan Reines <kgryte@gmail.com>

kgryte committed:
build: disable workflow

kgryte committed:
fix: address lint error

gunjjoshi committed:
feat: add C implementation for `math/base/special/gamma-lanczos-sum`

PR-URL: 	https://github.com/stdlib-js/stdlib/pull/1834
Closes: https://github.com/stdlib-js/stdlib/issues/1822
Ref: https://github.com/stdlib-js/stdlib/issues/649
Co-authored-by: Athan Reines <kgryte@gmail.com>
Reviewed-by: Athan Reines <kgryte@gmail.com>

AuenKr committed:
refactor: update `blas/ext/base/scusum` to follow current project conventions

PR-URL: 	https://github.com/stdlib-js/stdlib/pull/1864
Closes: https://github.com/stdlib-js/stdlib/issues/1516
Ref: https://github.com/stdlib-js/stdlib/issues/1152
Co-authored-by: Athan Reines <kgryte@gmail.com>
Reviewed-by: Athan Reines <kgryte@gmail.com>

stdlib-bot committed:
docs: update REPL namespace documentation

PR-URL: #1972

Signed-off-by: stdlib-bot <82920195+stdlib-bot@users.noreply.github.com>
Co-authored-by: Athan Reines <kgryte@gmail.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

rxbryan committed:
refactor: update `blas/ext/base/scusumors`

PR-URL: #1945
Closes: #1519

---------

Signed-off-by: Bryan Elee <rxbryn@gmail.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

rxbryan committed:
refactor: update `blas/ext/base/scusumpw` to follow current project conventions

PR-URL: #1929
Closes: #1520

---------

Signed-off-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
Co-authored-by: Pranav <85227306+Pranavchiku@users.noreply.github.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Pranav Goswami <goswami.4@iitj.ac.in>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

stdlib-bot committed:
docs: update list of contributors

PR-URL: #1974

Signed-off-by: stdlib-bot <82920195+stdlib-bot@users.noreply.github.com>
Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com> 
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

kgryte committed:
fix: update build configuration

Ref: https://github.com/stdlib-js/stdlib/pull/1929

kgryte committed:
fix: update build configuration

Ref: https://github.com/stdlib-js/stdlib/pull/1945
Ref: https://github.com/stdlib-js/stdlib/commit/548f903780b33e672074ce77fdaceae672679ef6

naveen1m committed:
refactor: update `blas/ext/base/dapx` to follow current project conventions

PR-URL: 	https://github.com/stdlib-js/stdlib/pull/1954
Closes: https://github.com/stdlib-js/stdlib/issues/1464
Ref: https://github.com/stdlib-js/stdlib/issues/1152
Co-authored-by: Athan Reines <kgryte@gmail.com>
Reviewed-by: Athan Reines <kgryte@gmail.com> 
Signed-off-by: Athan Reines <kgryte@gmail.com>

gunjjoshi committed:
refactor: update `blas/ext/base/sapxsum` to follow current project conventions

PR-URL: 	https://github.com/stdlib-js/stdlib/pull/1850
Closes: https://github.com/stdlib-js/stdlib/issues/1510
Ref: https://github.com/stdlib-js/stdlib/issues/1152
Co-authored-by: Athan Reines <kgryte@gmail.com>
Co-authored-by: Pranav Goswami <goswami.4@iitj.ac.in>
Reviewed-by: Athan Reines <kgryte@gmail.com>
Reviewed-by: Pranav Goswami <goswami.4@iitj.ac.in>
Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>
Signed-off-by: Pranav Goswami <goswami.4@iitj.ac.in>
Signed-off-by: Athan Reines <kgryte@gmail.com>

stdlib-bot committed:
docs: update namespace TypeScript declarations

PR-URL: 	https://github.com/stdlib-js/stdlib/pull/1975
Co-authored-by: Athan Reines <kgryte@gmail.com>
Reviewed-by: Athan Reines <kgryte@gmail.com> 
Signed-off-by: stdlib-bot <82920195+stdlib-bot@users.noreply.github.com>

stdlib-bot committed:
docs: update REPL namespace documentation

PR-URL: 	https://github.com/stdlib-js/stdlib/pull/1973
Co-authored-by: Athan Reines <kgryte@gmail.com>
Reviewed-by: Athan Reines <kgryte@gmail.com> 
Signed-off-by: stdlib-bot <82920195+stdlib-bot@users.noreply.github.com>

kgryte committed:
fix: update build configuration and remove unnecessary cast

Ref: #1861

kgryte committed:
style: fix indentation

Ref: #1861

rxbryan committed:
refactor: update `blas/ext/base/sapx` to follow current project conventions

PR-URL: 	https://github.com/stdlib-js/stdlib/pull/1824
Closes: https://github.com/stdlib-js/stdlib/issues/1509
Ref: https://github.com/stdlib-js/stdlib/issues/1152
Co-authored-by: Athan Reines <kgryte@gmail.com>
Reviewed-by: Athan Reines <kgryte@gmail.com> 
Signed-off-by: Athan Reines <kgryte@gmail.com>

kgryte committed:
fix: remove trailing commas

kgryte committed:
fix: use correct variable names and fix configuration file

kgryte committed:
refactor: remove explicit cast

kgryte committed:
refactor: remove explicit cast

kgryte committed:
refactor: remove explicit cast

gunjjoshi committed:
fix: rename include directory for `constants/float32/half-pi`

PR-URL: 	https://github.com/stdlib-js/stdlib/pull/1976
Ref: https://github.com/stdlib-js/stdlib/pull/1953
Reviewed-by: Athan Reines <kgryte@gmail.com>

stdlib-bot committed:
docs: update REPL namespace documentation

PR-URL: 	https://github.com/stdlib-js/stdlib/pull/1977
Co-authored-by: Athan Reines <kgryte@gmail.com>
Reviewed-by: Athan Reines <kgryte@gmail.com> 
Signed-off-by: stdlib-bot <82920195+stdlib-bot@users.noreply.github.com>

gunjjoshi committed:
feat: add C implementation for gamma-lanczos-sum-expg-scaled

gunjjoshi committed:
feat: add C implementation for gamma-lanczos-sum-expg-scaled

gunjjoshi committed:
Revert "Update test.native.js"

This reverts commit dbfeea7fdb14cad40605174668065852cab5a486.

gunjjoshi commented:
I will be opening a new pull request for this one, everything is getting mixed up here. Closing this for now.
